cmd/binlog-collector/storage/storage.go

```diff
@@ -0,0 +1,60 @@
+package storage
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/pkg/errors"
+)
+
+type Client interface {
+	GetObject(objectName string) ([]byte, error)
+	PutObject(name string, data io.Reader) error
+}
+
+// S3 is a type for working with S3 storages
+type S3 struct {
+	minioClient *minio.Client   // minio client for work with storage
+	ctx         context.Context // context for client operations
+	bucketName  string          // S3 bucket name where binlogs will be stored
+}
+
+// NewS3 return new Manager, useSSL using ssl for connection with storage
+func NewS3(endpoint, accessKeyID, secretAccessKey, bucketName string, useSSL bool) (*S3, error) {
+	minioClient, err := minio.New(endpoint, &minio.Options{
+		Creds:  credentials.NewStaticV2(accessKeyID, secretAccessKey, ""),
+		Secure: useSSL,
+	})
```

Comment:
Region must be from CR also
```suggestion
	minioClient, err := minio.NewWithRegion(endpoint, accessKeyID, secretAccessKey, useSSL, region)
```

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/622/commits/cc36d3efa9217d24a890b91613033638fc2ea139